zmq-multiplex.slide

```diff
@@ -0,0 +1,136 @@
+ZMQ multiplex between golang coroutine and python thread
+10 June 2017
+
+Rick Mak
+Oursky
+rickmak@oursky.com
+@rickmak
+CC81327D
+
+* What we want to do
+
+- Enable building of scalable software with less learning curve
+- Provide foundation for developer to focus business requirement and UI
+- Save developer from deployment hassle
+
+* Skygear-server
+
+- Design for web development
+- Written in golang
+- Provide basic function like Auth, Record CRUD, PubSub via core
+- Provide plugin for extending function, fulfilling business requirement.
+
+* ZMQ
+
+- High performance messaging queue
+- No message broker
+
+* Python
+
+- Imperative
+- Easy to write
+- Easy to read
+- Code quickly
+
+Actually we have JS too.
+
+* Data flow
+
+.image zmq-multiplex/dataflow.png
+
+- Skygear-server is written in golang and able to handle thousands of requests
+- For request triggering cloud code, a message will send to worker via zmq
+
+* Wire Protocol
+
+    [WORKER-ADDR|0|body....]
+
+- Simple
+- Routing based on WORKER-ADDR
+
+The protocol also have heartbeat, details:
+https://rfc.zeromq.org/spec:6/PPP
+
+* Problem
+
+- Python-side resource exhausted on high load
+- Thread overhead is much higher than goroutine
+- Recursion, cloud code will call back skygear-server
+- User don't want to handle the resource problem
+- Providing a “just works” solution
```

Comment:
I think this list has everything you intend to cover, but the list is not written in the way you will convey the idea by speech...let me come around and here you say it so I can give more suggestion.

This makes a good article.